Don't simultaneously test -c and -v switches

jkeenan opened the pull request:
Up through perl-5.37.3, there was a bug in perl (see https://github.com/Perl/perl5/issues/20252#issuecomment-1236880872) which permitted simultaneous use of the '-c' and '-v' switches to the perl interpreter.  Once this bug was corrected, t/01all.t began to fail. As Test::Strict has many CPAN distributions depending on it, this test failure has had wide impact.

This pull request modifies tests appropriately and has been tested on bleadperl (v5.37.7-134-g25948dfb24) and on perl-5.32.1, in both cases on threaded builds on FreeBSD-12.  Devel::Cover was not installed on bleadperl, hence t/04cover.t was skipped on that build.  Otherwise, all tests are passing.  In the course of testing, certain additions were made to .gitignore and MANIFEST.SKIP to get metadata-oriented tests to pass.

NOTE:  In my git checkout I find the following file:

        .git/hooksfsmonitor-watchman.sample

This is deemed by 'file' to be a Perl executable.  As such, it is pushed onto the list of files tested by 'all_perl_files_ok()' at the start of t/01all.t.  This in turn means that in my 'git checkout' the count of tests is off by 2 and the test is reported to fail.

        "Looks like you planned 59 tests but ran 61."

Because the files in .git/hooks/ should not be included in the tarball, the count of tests is correct when run from the tarball.  I leave to the maintainer to determine how to square this circle.  You should probably check for the presence of a .git/ directory and exclude that directory from the list of directories passed to all_perl_files_ok().

For https://github.com/manwar/Test-Strict/issues/32

jkeenan committed:
Don't simultaneously test -c and -v switches

Up through perl-5.37.3, there was a bug in perl (see
https://github.com/Perl/perl5/issues/20252#issuecomment-1236880872)
which permitted simultaneous use of the '-c' and '-v' switches to the
perl interpreter.  Once this bug was corrected, t/01all.t began to fail.
As Test::Strict has many CPAN distributions depending on it, this test
failure has had wide impact.

This pull request modifies tests appropriately and has been tested on
bleadperl (v5.37.7-134-g25948dfb24) and on perl-5.32.1, in both cases on
threaded builds on FreeBSD-12.  Devel::Cover was not installed on
bleadperl, hence t/04cover.t was skipped on that build.  Otherwise, all
tests are passing.  In the course of testing, certain additions were
made to .gitignore and MANIFEST.SKIP to get metadata-oriented tests to
pass.

NOTE:  In my git checkout I find the following file:

        .git/hooksfsmonitor-watchman.sample

This is deemed by 'file' to be a Perl executable.  As such, it is pushed
onto the list of files tested by 'all_perl_files_ok()' at the start of
t/01all.t.  This in turn means that in my 'git checkout' the count of
tests is off by 2 and the test is reported to fail.

        "Looks like you planned 59 tests but ran 61."

Because the files in .git/hooks/ should not be included in the tarball,
the count of tests is correct when run from the tarball.  I leave to the
maintainer to determine how to square this circle.  You should probably
check for the presence of a .git/ directory and exclude that directory
from the list of directories passed to all_perl_files_ok().

For https://github.com/manwar/Test-Strict/issues/32

demerphq commented:
Yeah, this looks right to me. Passes tests as well, except for the issue you mentioned about "Looks like you planned 59 tests but ran 61."

bbpeterharrison commented:
Because you've removed the "exit" statement in the test, you're running two more tests.
Either a) put the exit back (not ideal)or
b) Increase the number of tests to run by 2 on line 18.

jkeenan commented:
> Because you've removed the "exit" statement in the test, you're running two more tests. Either a) put the exit back (not ideal)or b) Increase the number of tests to run by 2 on line 18.

It's not that simple.  As I pointed out in my original posting last year, the count of files to be tested by `all_perl_files_ok()` is different depending on whether you are in a `git` checkout or an unwrapped tarball.

Just now I took your suggestion and added `2` to the plan in `t/01all.t`.  Test PASSed in the `git` checkout.  I called `make dist`, moved the tarball to a tempdir, unwrapped, built and tested it.  Test FAILed.  I offered @manwar some options.  He needs to choose one of those options, get the tests passing both in checkout and from tarball, and issue a new CPAN release.

manwar commented:
@jkeenan Apology for late response. I will get this sorted asap.

manwar commented:
@jkeenan Please correct me if I am missing the point. I tested your changes and still having the same issue.
<img width="727" alt="image" src="https://github.com/user-attachments/assets/33a07428-d0da-43cd-a062-7b2d33981c87">

haarg commented:
I can't reproduce that problem with this PR.

Can you show the full test log?

manwar commented:
@haarg here is the complete test log:
<img width="1240" alt="image" src="https://github.com/user-attachments/assets/3a4c13a8-ca4b-4418-b5cc-45fc21245747">

haarg commented:
Can you run `perl -Ilib t/01all.t` and show the output?

manwar commented:
@haarg here is the log:
<img width="437" alt="image" src="https://github.com/user-attachments/assets/a659a11e-3188-46f1-9c53-2e088bd22f23">

haarg commented:
That looks like it should be fixed in you merge both this and #34.

manwar commented:
Merged the pull request manually, thanks @jkeenan
